### PR TITLE
Send server timestamp along with KPI data.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
- Add tests to make sure all expected fields are present.

issue #1660

### CODE REVIEW

File: resources/static/test/cases/shared/models/interaction_data.js [hunk 1]
```diff
@@ -87,7 +87,16 @@
...
-        model.push({ lang: "bar", secret: "Attack at dawn!!!" });
+        var now = new Date().getTime();
+
+        model.push({
+          event_stream: [],
+          sample_rate: 1,
+          timestamp: now,
+          local_timestamp: now,
+          lang: "bar",
+          secret: "Attack at dawn!!!"
```
Comment (lloyd) [inline resources/static/test/cases/shared/models/interaction_data.js:98]: what is this secret?  what's its utility and intended configuration in production?

File: resources/static/test/cases/shared/models/interaction_data.js [hunk 2]
```diff
@@ -87,7 +87,16 @@
 
         // When the interaction_data next completes, this will be the only data
         // that is pushed.
-        model.push({ lang: "bar", secret: "Attack at dawn!!!" });
+        var now = new Date().getTime();
+
+        model.push({
+          event_stream: [],
+          sample_rate: 1,
+          timestamp: now,
```
Comment (lloyd) [inline resources/static/test/cases/shared/models/interaction_data.js:95]: didn't we agree to round timestamp down to 5 minute granularity to make log correlation intractable, even for ourselves?  is this the work of a subsequent bug?  

We should do this in browserid before blobs leave for the piggy bank.

File: resources/static/test/cases/shared/models/interaction_data.js (hunk 1 above)
Comment (shane-tomlinson) [inline resources/static/test/cases/shared/models/interaction_data.js:98]: @lloyd - it is an extraneous field to check to make sure the whitelist is working properly.

File: resources/static/test/cases/shared/models/interaction_data.js (hunk 2 above)
Comment (shane-tomlinson) [inline resources/static/test/cases/shared/models/interaction_data.js:95]: @lloyd - @jedp took care of the rounding on the backend.  We can do it on the front end as well.

Comment (lloyd) [inline resources/static/test/cases/shared/models/interaction_data.js:95]: I think the browserid router process is where we'd want to round, not in frontend code nor in kpi servers.

make sense?

**Pull request merged**